Firmware/first_lay_cal.cpp

```diff
@@ -163,88 +178,93 @@ void lay1cal_before_meander()
 //! @brief Print meander start
 void lay1cal_meander_start(float layer_height, float extrusion_width)
 {
-    enquecommand_P(PSTR("G1 X50 Y155"));
-
-    static const char fmt1[] PROGMEM = "G1 Z%-.3f F7200";
+#ifndef NEW_FIRST_LAYER_CAL
+    enquecommand_P(PSTR("G1X50Y155"));
+#endif //_NEW_FIRST_LAYER_CAL
+    static const char fmt1[] PROGMEM = "G1Z%g";
     enquecommandf_P(fmt1, layer_height);
-
-    enquecommand_P(PSTR("G1 F1080"));
-
-    enquecommandf_P(extrude_fmt,  75, 155, count_e(layer_height, extrusion_width * 4.f, 25));
-    enquecommandf_P(extrude_fmt, 100, 155, count_e(layer_height, extrusion_width * 2.f, 25));
-    enquecommandf_P(extrude_fmt, 200, 155, count_e(layer_height, extrusion_width, 100));
-    enquecommandf_P(extrude_fmt, 200, 135, count_e(layer_height, extrusion_width, 20));
+    enquecommand_P(feedrate_F1080);
+#ifdef NEW_FIRST_LAYER_CAL
+    enquecommandf_P(extrude_fmt_Y, short_length, count_e(layer_height, extrusion_width, short_length));
+#endif //_NEW_FIRST_LAYER_CAL
+
+    enquecommand_P(MSG_G91); //enable relative XYZ
+#ifndef NEW_FIRST_LAYER_CAL
+    enquecommandf_P(extrude_fmt_X, (float)25*invert, count_e(layer_height, extrusion_width * 4.f, 25));
+    enquecommandf_P(extrude_fmt_X, (float)25*invert, count_e(layer_height, extrusion_width * 2.f, 25));
+    enquecommandf_P(extrude_fmt_X, (float)100*invert, count_e(layer_height, extrusion_width, 100));
+    enquecommandf_P(extrude_fmt_Y, (float)-20*invert, count_e(layer_height, extrusion_width, 20));
+#else
+    enquecommandf_P(extrude_fmt_X, long_length*invert, count_e(layer_height, extrusion_width, long_length));
+    enquecommandf_P(extrude_fmt_Y, -short_length*invert, count_e(layer_height, extrusion_width, short_length));
+#endif //_NEW_FIRST_LAYER_CAL
 }
 
 //! @brief Print meander
 //! @param cmd_buffer character buffer needed to format gcodes
 void lay1cal_meander(float layer_height, float extrusion_width)
 {
-    const float short_length = 20;
-    float long_length = 150;
     const float long_extrusion = count_e(layer_height, extrusion_width, long_length);
     const float short_extrusion = count_e(layer_height, extrusion_width, short_length);
 
-    uint8_t y_pos = 135;
-    uint8_t x_pos = 50;
+    float x_pos = long_length;
+
     for(uint8_t i = 0; i <= 4; ++i)
     {
-        enquecommandf_P(extrude_fmt, x_pos, y_pos, long_extrusion);
-
-        y_pos -= short_length;
-
-        enquecommandf_P(extrude_fmt, x_pos, y_pos, short_extrusion);
+        enquecommandf_P(extrude_fmt_X, -x_pos*invert, long_extrusion);
 
-        x_pos += long_length;
+        x_pos = -x_pos;
 
-        long_length = -long_length;
+        enquecommandf_P(extrude_fmt_Y, -short_length*invert, short_extrusion);
     }
+#ifdef NEW_FIRST_LAYER_CAL
+    enquecommandf_P(extrude_fmt_X, -(long_length/2-square_width/2), long_extrusion); //~Middle of bed X125
```

Comment:
Consider using multiplication instead:

`long_length/2` -> `long_length * 0.5f`

I may be the compiler is smart enough to do this change automatically, but lets be explicit. Multiplication is more efficient than division.

Reply:
Woops, commented on the wrong thing.

This line is overextruded since a shorter length is used with the same E count. Needs adjusting

![image](https://github.com/user-attachments/assets/19e5a2a5-9399-4342-a6a7-9bd7f8cf67d4)

Reply:
@leptun Good point. the extrusion needs to be 0.5 of the long_length

Reply:
Are you sure? Feels to me like it would be better but still not 100% correct as that would still slightly overextrude. Remember that the line is (half of long_length - half of the square width). So just setting it to 0.5x will not account for the square size

Reply:
> Consider using multiplication instead:
> 
> `long_length/2` -> `long_length * 0.5f`
> 
> It may be the compiler is smart enough to do this change automatically, but lets be explicit. Multiplication is more efficient than division.

All divisions are done inside constexpr or on constant compile time computed variables, so I am not worried about these.

Reply:
@3d-gussner fixed overextrusion with new pattern. Please check

Reply:
Reminder for myself: check if that division is done at compile time. If not, is the division done once or twice.

Reply:
It was. I extracted that to a constexpr and no size change happened. After that I also switched to 0.5f*. Changed in https://github.com/prusa3d/Prusa-Firmware/pull/4811/commits/7a47bf30be79c6b9e9c042dbd6384eccad28a592